app/views/videos/show.html.haml

```diff
@@ -6,43 +6,36 @@
       .video_info.col-sm-3
         %header
           %h3= @video.title
-          %span Rating: 4.5/5.0
+          %span= "Rating: #{@video.average_rating}/5.0" if @reviews.count > 0
         %p= @video.description
         .actions
           %a.btn.btn-primary(href="") Watch Now
-          %a.btn.btn-default(href="") + My Queue
+          = link_to "+ My Queue", queue_items_path(video_id: @video.id), class: "btn btn-default", method: "post"
 
 %section.reviews.container
   .row
     .col-sm-10.col-sm-offset-1
-      %form
+      = bootstrap_form_for [@video, @review] do |f|
         %fieldset
-          .form-group
-            %label Rate this video
-            .row
-              .col-sm-3
-                %select.form-control(name="")
-                  %option(value="5") 5 Stars
-                  %option(value="4") 4 Stars
-                  %option(value="3") 3 Stars
-                  %option(value="2") 2 Stars
-                  %option(value="1") 1 Star
-          .form-group
-            %label Write Review
-            .row
-              .col-sm-8
-                %textarea.form-control(name="" rows="6")
-        %fieldset.form-group.actions.clearfix
-          %input(type="submit" value="Submit" class="btn")
-          %a(href="") Cancel
+          .row
+            .col-sm-3
+              = f.select(:rating, [["5 Stars", 5], ["4 Stars", 4], ["3 Stars", 3], ["2 Stars", 2], ["1 Star", 1]], { label: "Rate this video" })
+          .row
+            .col-sm-8
+              = f.text_area :description, label: "Write Review", rows: 6
+        %fieldset.actions.clearfix
+          = f.submit "Submit", class: "btn"
+          = link_to "Cancel", video_path(@video)
       %header
-        %h3 User Reviews (253)
+        %h3= "User Reviews (#{@reviews.count})"
       %ul
-        - 8.times do
+        - @reviews.reload.each do |review|
```

Comment:
Definitely not a recommended way to handle this here, because for @reviews where you don't need to reload, you just query database again. So this should be done only in controller and only where you need it.